src/mindlessgen/molecules/miscellaneous.py

```diff
@@ -33,9 +34,20 @@ def set_random_charge(ati: np.ndarray, verbosity: int = 1) -> tuple[int, int]:
                 ln_protons += (
                     atom - 3 + 1
                 )  # subtract 3 to get the number of protons in the Ln3+ ion
-        ligand_protons = nel - ln_protons
+            elif atom in get_actinides():
+                if atom < 96:
+                    uhf += atom - 88
+                else:
+                    uhf += 102 - atom
+                ac_protons += (
+                    atom - 3 + 1
+                )  # subtract 3 to get the number of protons in the Ln3+ ion
+        ligand_protons = nel - ln_protons - ac_protons
```

Comment:
Is there a test for this algorithm? If not, we should implement one.

Reply:
Add the lanthanide tests just here:
`test/test_molecules/test_miscellaneous.py`